src/elements/select/select.spec.ts

```diff
@@ -1174,6 +1159,135 @@ describe(`sbb-select`, () => {
+      await openSpy.calledOnce();
+
+      firstOption.click();
+      await closeSpy.calledOnce();
+      await waitForLitRender(element);
+
+      expect(element.getDisplayValue()).to.be.equal('Yes');
+      expect(firstOption).to.have.attribute('selected');
+      expect(element.value).to.be.equal(true);
+    });
+  });
+
+  describe('with number value', () => {
+    let element: SbbSelectElement<number>;
+
+    beforeEach(async () => {
+      const root = await fixture(
+        html`<form>
+          <sbb-form-field>
+            <label>Label</label>
+            <sbb-select placeholder="Placeholder" name="select1" .value=${0}>
+              <sbb-option id="option-1" .value=${0}>No</sbb-option>
+              <sbb-option id="option-2" .value=${1}>Yes</sbb-option>
+            </sbb-select>
+          </sbb-form-field>
+        </form>`,
+      );
+      element = root.querySelector<SbbSelectElement<number>>('sbb-select')!;
+    });
+
+    it('should have the correct displayValue and the first option selected', () => {
+      const firstOption = element.querySelector<SbbOptionElement<boolean>>('#option-1')!;
+      expect(element.getDisplayValue()).to.be.equal('No');
+      expect(firstOption).to.have.attribute('selected');
+    });
+
+    it('should set value on option click', async () => {
+      const secondOption = element.querySelector<SbbOptionElement<boolean>>('#option-2')!;
+      const openSpy = new EventSpy(SbbSelectElement.events.open, element);
+      const closeSpy = new EventSpy(SbbSelectElement.events.close, element);
+
+      element.click();
+      await openSpy.calledOnce();
+
+      secondOption.click();
+      await closeSpy.calledOnce();
+      await waitForLitRender(element);
+
+      expect(element.getDisplayValue()).to.be.equal('Yes');
+      expect(secondOption).to.have.attribute('selected');
+      expect(element.value).to.be.equal(1);
+    });
+  });
+
+  describe('with null value', () => {
+    let element: SbbSelectElement<null | 0>;
+
+    beforeEach(async () => {
+      const root = await fixture(
+        html`<form>
+          <sbb-form-field>
+            <label>Label</label>
+            <sbb-select placeholder="Placeholder" name="select1" .value=${null}>
+              <sbb-option id="option-1" .value=${0}>other</sbb-option>
+              <sbb-option id="option-2" .value=${null}>null</sbb-option>
+            </sbb-select>
+          </sbb-form-field>
+        </form>`,
+      );
+      element = root.querySelector<SbbSelectElement<null | 0>>('sbb-select')!;
+    });
+
+    it('should have the correct displayValue and the first option selected', () => {
+      const secondOption = element.querySelector<SbbOptionElement<boolean>>('#option-2')!;
+      expect(element.getDisplayValue()).to.be.equal('null');
+      expect(secondOption).to.have.attribute('selected');
+    });
+
+    it('should set value on option click', async () => {
+      element.value = 0;
+      await waitForLitRender(element);
+
```

Comment:
I would check here that the valueToOptions logic is working =>
expect displayValue to be 0
expect firstOption to be selected